## VR Norm

In [1]:
#Data saved in csv/excel
import glob, os
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
reg = pd.read_csv('C:/Users/tsb7592/Downloads/registration_data_distinct.csv',index_col=0)

In [3]:
reg.shape

(3260, 13)

In [171]:
df = pd.read_csv('C:/Users/tsb7592/Downloads/VR_norm.csv', index_col=0)

# OR Norm & Original Norm

In [172]:
df =df[~df['ItemID'].str.contains('Practice|Title|Intro|Transition')]

In [173]:
df.shape

(924010, 11)

In [174]:
#Importing the FormWords
xls =  pd.read_excel('C:/Users/tsb7592/Downloads/OR Forms for Norming 07.08.2020_WithWords.xlsx', sheet_name = "Formatted_Forms")

In [175]:
df.Key.unique()

array(['InstrumentOrder', 'InstrumentSection', 'ItemOrder', 'Locale',
       'Response', 'Score', 'DataType', 'Position', 'ResponseTime',
       'AppVersion', 'iPadVersion', 'FirmwareVersion'], dtype=object)

In [176]:
#Select the 3 needed key values
df1 =df[(df.Key == 'Response')]
df1['Value'] =pd.to_numeric(df1['Value'])

In [177]:
df2 =df[(df.Key == 'ResponseTime')]
df2['Value'] =pd.to_numeric(df2['Value'])

In [178]:
df3 =df[(df.Key == 'Score')]
df3['Value'] =pd.to_numeric(df3['Value'])

In [179]:
df2.Key.value_counts()

ResponseTime    77000
Name: Key, dtype: int64

In [180]:
df1 = df1[['PIN','RegistrationID','ItemID','Key','Value']].drop_duplicates()
df2 = df2[['PIN','RegistrationID','ItemID','Key','Value']].drop_duplicates()
df3 = df3[['PIN','RegistrationID','ItemID','Key','Value']].drop_duplicates()

In [181]:
df1.shape

(75501, 5)

In [182]:
correct = df3[df3.Value == 1]

In [183]:
wrong= df3[df3.Value == 0]

In [184]:
correctppl =pd.merge(correct, df2, on=['ItemID','PIN'] )
wrongppl =pd.merge(wrong, df2, on=['ItemID','PIN'] )

In [185]:
correctppl.columns =['PIN', 'RegistrationID', 'ItemID', 'Key_x', 'Value_x',
       'RegistrationID_y', 'Key_y', 'Time']
wrongppl.columns =['PIN', 'RegistrationID', 'ItemID', 'Key_x', 'Value_x',
       'RegistrationID_y', 'Key_y', 'Time']


In [186]:
cp = correctppl.pivot_table(index = ['PIN', 'RegistrationID'], columns='ItemID',values='Time').reset_index()

In [187]:
wp = wrongppl.pivot_table(index = ['PIN', 'RegistrationID'], columns='ItemID',values='Time').reset_index()
wp

ItemID        PIN                        RegistrationID  VA_RA022  VR_MR001  \
0             009  ABCE632D-1895-4DD5-9773-E848E3A1B396       NaN       NaN   
1        12345678  BE7A8565-9699-42D5-BA5F-86D304B50BA4       NaN       NaN   
2         5NA6010  8EE1F715-6AE0-43EC-B053-1EAB3BF9E1D0       NaN       NaN   
3        AL Train  E0DA4239-102F-4B79-9775-86DE3C1BD6BE       NaN       NaN   
4       ATLCERT11  1E28145A-D99E-4FBF-B253-D21EB86445EA       NaN       NaN   
...           ...                                   ...       ...       ...   
3493     msl50001  E51171E2-F00C-4B48-885B-5A08C6E5190C       NaN       NaN   
3494     msl50005  E2E3E3EE-D085-4513-85C5-8EFE10EA9AEE       NaN       NaN   
3495     sna40104  B22BD1C5-C330-44FA-96F7-41CF7425970D       NaN       NaN   
3496     wPX22103  AC5AC37B-4AB4-433A-BAB7-073B36DAD54F       NaN       NaN   
3497     wpx22106  F215A453-F8D7-48B6-8FD1-E767D12CFC41       NaN       NaN   

ItemID  VR_MR002  VR_MR003  VR_MR004  VR_MR005  VR_MR006   VR_MR007  ...  \
0            NaN       NaN    2.6395       NaN       NaN   3.721695  ...   
1            NaN   33.9842       NaN       NaN       NaN  19.996480  ...   
2            NaN       NaN       NaN       NaN       NaN        NaN  ...   
3            NaN       NaN       NaN       NaN       NaN        NaN  ...   
4       21.67658       NaN       NaN       NaN       NaN        NaN  ...   
...          ...       ...       ...       ...       ...        ...  ...   
3493         NaN       NaN       NaN       NaN       NaN  15.105690  ...   
3494         NaN       NaN       NaN       NaN       NaN        NaN  ...   
3495         NaN       NaN       NaN       NaN       NaN        NaN  ...   
3496         NaN       NaN       NaN       NaN       NaN        NaN  ...   
3497         NaN       NaN       NaN  23.55377  24.87863        NaN  ...   

ItemID  VR_SV003  VR_SV004R1  VR_SV005R1  VR_SV006R1  VR_SV007R1  VR_SV008R1  \
0            NaN    3.371989         NaN         NaN         NaN         NaN   
1            NaN         NaN         NaN         NaN         NaN         NaN   
2            NaN         NaN         NaN         NaN         NaN         NaN   
3            NaN         NaN         NaN         NaN         NaN         NaN   
4            NaN         NaN         NaN         NaN         NaN         NaN   
...          ...         ...         ...         ...         ...         ...   
3493         NaN         NaN         NaN         NaN    8.966269         NaN   
3494         NaN         NaN         NaN         NaN         NaN         NaN   
3495         NaN         NaN         NaN    1.656719         NaN    26.10057   
3496         NaN         NaN         NaN         NaN         NaN         NaN   
3497         NaN         NaN         NaN         NaN         NaN         NaN   

ItemID  VR_SV009  VR_SV010  VR_SV011  VR_SV012  
0            NaN       NaN       NaN       NaN  
1            NaN       NaN       NaN       NaN  
2            NaN       NaN       NaN       NaN  
3            NaN       NaN  11.35671       NaN  
4            NaN  28.10999       NaN       NaN  
...          ...       ...       ...       ...  
3493         NaN       NaN       NaN       NaN  
3494         NaN       NaN       NaN       NaN  
3495         NaN       NaN       NaN       NaN  
3496    34.43225       NaN       NaN       NaN  
3497         NaN       NaN       NaN       NaN  

[3498 rows x 196 columns]

In [188]:
vr1=pd.merge(df1, reg[['PIN','Age']], on=['PIN'], how='left')
vr2=pd.merge(df2, reg[['PIN','Age']], on=['PIN'], how='left')
vr3=pd.merge(df3, reg[['PIN','Age']], on=['PIN'], how='left')

In [189]:
vr1.ItemID.nunique()

194

In [190]:
#long to wide
VR_Response =vr1.pivot_table(index = ['PIN', 'RegistrationID','Age'], columns='ItemID',values='Value').reset_index()

In [191]:
VR_ResponseTime =vr2.pivot_table(index = ['PIN','RegistrationID', 'Age'], columns='ItemID',values='Value').reset_index()

In [192]:
VR_Score =vr3.pivot_table(index = ['PIN', 'RegistrationID','Age'], columns='ItemID',values='Value').reset_index()

In [193]:
VR_ResponseTime.to_csv('C:/Users/tsb7592/Downloads/VR_ResponseTime.csv',index=False)

In [194]:
VR_Response.to_csv('C:/Users/tsb7592/Downloads/VR_Response.csv',index=False)

In [195]:
VR_Score.to_csv('C:/Users/tsb7592/Downloads/VR_Score.csv',index=False)

# Analysis
## Reaction time: 


In [196]:
VR_Score.VR_MR001

0       NaN
1       NaN
2       NaN
3       NaN
4       NaN
       ... 
3157    NaN
3158    0.0
3159    NaN
3160    NaN
3161    NaN
Name: VR_MR001, Length: 3162, dtype: float64

In [197]:
df[(df.ItemID == 'VR_MR001') ].shape

(7920, 11)

In [198]:
VR_Score.describe()

ItemID          Age    VA_RA022    VR_MR001    VR_MR002    VR_MR003  \
count   3162.000000  128.000000  581.000000  410.000000  358.000000   
mean      25.553763    0.226562    0.524957    0.390244    0.357542   
std       22.354449    0.420252    0.499807    0.488401    0.479947   
min        3.000000    0.000000    0.000000    0.000000    0.000000   
25%       10.000000    0.000000    0.000000    0.000000    0.000000   
50%       16.000000    0.000000    1.000000    0.000000    0.000000   
75%       37.000000    0.000000    1.000000    1.000000    1.000000   
max       96.000000    1.000000    1.000000    1.000000    1.000000   

ItemID    VR_MR004    VR_MR005    VR_MR006    VR_MR007    VR_MR008  ...  \
count   409.000000  424.000000  394.000000  389.000000  375.000000  ...   
mean      0.266504    0.389151    0.451777    0.344473    0.088000  ...   
std       0.442672    0.488134    0.498302    0.475808    0.283673  ...   
min       0.000000    0.000000    0.000000    0.000000    0.000000  ...   
25%       0.000000    0.000000    0.000000    0.000000    0.000000  ...   
50%       0.000000    0.000000    0.000000    0.000000    0.000000  ...   
75%       1.000000    1.000000    1.000000    1.000000    0.000000  ...   
max       1.000000    1.000000    1.000000    1.000000    1.000000  ...   

ItemID    VR_SV003  VR_SV004R1  VR_SV005R1  VR_SV006R1  VR_SV007R1  \
count   373.000000  589.000000  407.000000  437.000000  376.000000   
mean      0.453083    0.679117    0.594595    0.441648    0.380319   
std       0.498463    0.467213    0.491575    0.497152    0.486112   
min       0.000000    0.000000    0.000000    0.000000    0.000000   
25%       0.000000    0.000000    0.000000    0.000000    0.000000   
50%       0.000000    1.000000    1.000000    0.000000    0.000000   
75%       1.000000    1.000000    1.000000    1.000000    1.000000   
max       1.000000    1.000000    1.000000    1.000000    1.000000   

ItemID  VR_SV008R1    VR_SV009    VR_SV010    VR_SV011    VR_SV012  
count   385.000000  404.000000  398.000000  388.000000  389.000000  
mean      0.093506    0.373762    0.402010    0.365979    0.634961  
std       0.291520    0.484402    0.490921    0.482326    0.482061  
min       0.000000    0.000000    0.000000    0.000000    0.000000  
25%       0.000000    0.000000    0.000000    0.000000    0.000000  
50%       0.000000    0.000000    0.000000    0.000000    1.000000  
75%       0.000000    1.000000    1.000000    1.000000    1.000000  
max       1.000000    1.000000    1.000000    1.000000    1.000000  

[8 rows x 195 columns]

In [199]:
##For each item, how long did it take people on average to complete the item? 
count =VR_ResponseTime.describe().round(2).iloc[0]
meantime =VR_ResponseTime.describe().round(2).iloc[1]

In [200]:
rv = pd.concat([count,meantime], axis=1)
rv =rv.iloc[1:]

In [201]:
rv.columns =['N', 'MeanTime_Overall']

In [202]:
rv =rv.reset_index()

In [203]:
timecol = VR_ResponseTime.columns[3:].to_list()

In [204]:
timecol

['VA_RA022',
 'VR_MR001',
 'VR_MR002',
 'VR_MR003',
 'VR_MR004',
 'VR_MR005',
 'VR_MR006',
 'VR_MR007',
 'VR_MR008',
 'VR_MR009R1',
 'VR_MR010',
 'VR_MR011R1',
 'VR_MR012',
 'VR_MR013R1',
 'VR_MR014R1',
 'VR_MR015R1',
 'VR_MR016',
 'VR_MR017',
 'VR_MR018',
 'VR_MR019R1',
 'VR_MR020',
 'VR_MR021',
 'VR_MR022',
 'VR_MR023',
 'VR_MR024',
 'VR_MR025R1',
 'VR_MR026R1',
 'VR_MR027R1',
 'VR_MR028',
 'VR_MR029',
 'VR_MR030R1',
 'VR_MR031R1',
 'VR_MR032',
 'VR_MR033R1',
 'VR_MR034R1',
 'VR_MR035',
 'VR_MR036',
 'VR_MR037',
 'VR_MR038',
 'VR_MR039R1',
 'VR_PC001R1',
 'VR_PC002',
 'VR_PC004R1',
 'VR_PC005R1',
 'VR_PC006R1',
 'VR_PC007',
 'VR_PC008',
 'VR_PC009',
 'VR_PC011',
 'VR_PC012R1',
 'VR_PC013',
 'VR_PC014R1',
 'VR_PC015',
 'VR_PC016',
 'VR_PC017',
 'VR_PC018R1',
 'VR_PC019R1',
 'VR_PC020',
 'VR_PC021R1',
 'VR_PC022R1',
 'VR_PC023',
 'VR_PC024',
 'VR_PC025',
 'VR_PC026',
 'VR_PC027',
 'VR_RA001',
 'VR_RA002R1',
 'VR_RA004',
 'VR_RA011R1',
 'VR_RA013',
 'VR_RA014R1',
 'VR_RA015R1',
 'VR_RA0

In [205]:
adc ={}
for col in timecol:
    adc[col]=round(VR_ResponseTime[VR_ResponseTime[col].notnull()]['Age'].mean(),2)

In [206]:
meanage = pd.DataFrame.from_dict(adc,orient='index')

In [207]:
meanage=meanage.reset_index()
meanage.columns=['ItemID','mean_age']

In [208]:
meanage

ItemID  mean_age
0      VA_RA022      3.52
1      VR_MR001     24.82
2      VR_MR002     32.84
3      VR_MR003     30.00
4      VR_MR004     32.21
..          ...       ...
189  VR_SV008R1     33.39
190    VR_SV009     32.80
191    VR_SV010     32.66
192    VR_SV011     30.05
193    VR_SV012     31.32

[194 rows x 2 columns]

In [209]:
data=pd.merge(meanage, rv, on="ItemID")

In [210]:
data.head()

ItemID  mean_age      N  MeanTime_Overall
0  VA_RA022      3.52  128.0              7.05
1  VR_MR001     24.82  581.0             11.39
2  VR_MR002     32.84  410.0             18.37
3  VR_MR003     30.00  358.0             19.06
4  VR_MR004     32.21  409.0             17.57


Within each item, how long did it take people on average to complete the item if they got it right, and if they got it wrong? 

So, the resulting dataset should be approx 195 rows and 9 columns. For the 6th column you may need to join the Response dataset you created earlier to get this column, so maybe do it at the very end. 

In [211]:
VR_Score.describe().round(2)

ItemID      Age  VA_RA022  VR_MR001  VR_MR002  VR_MR003  VR_MR004  VR_MR005  \
count   3162.00    128.00    581.00    410.00    358.00    409.00    424.00   
mean      25.55      0.23      0.52      0.39      0.36      0.27      0.39   
std       22.35      0.42      0.50      0.49      0.48      0.44      0.49   
min        3.00      0.00      0.00      0.00      0.00      0.00      0.00   
25%       10.00      0.00      0.00      0.00      0.00      0.00      0.00   
50%       16.00      0.00      1.00      0.00      0.00      0.00      0.00   
75%       37.00      0.00      1.00      1.00      1.00      1.00      1.00   
max       96.00      1.00      1.00      1.00      1.00      1.00      1.00   

ItemID  VR_MR006  VR_MR007  VR_MR008  ...  VR_SV003  VR_SV004R1  VR_SV005R1  \
count     394.00    389.00    375.00  ...    373.00      589.00      407.00   
mean        0.45      0.34      0.09  ...      0.45        0.68        0.59   
std         0.50      0.48      0.28  ...      0.50        0.47        0.49   
min         0.00      0.00      0.00  ...      0.00        0.00        0.00   
25%         0.00      0.00      0.00  ...      0.00        0.00        0.00   
50%         0.00      0.00      0.00  ...      0.00        1.00        1.00   
75%         1.00      1.00      0.00  ...      1.00        1.00        1.00   
max         1.00      1.00      1.00  ...      1.00        1.00        1.00   

ItemID  VR_SV006R1  VR_SV007R1  VR_SV008R1  VR_SV009  VR_SV010  VR_SV011  \
count       437.00      376.00      385.00    404.00    398.00    388.00   
mean          0.44        0.38        0.09      0.37      0.40      0.37   
std           0.50        0.49        0.29      0.48      0.49      0.48   
min           0.00        0.00        0.00      0.00      0.00      0.00   
25%           0.00        0.00        0.00      0.00      0.00      0.00   
50%           0.00        0.00        0.00      0.00      0.00      0.00   
75%           1.00        1.00        0.00      1.00      1.00      1.00   
max           1.00        1.00        1.00      1.00      1.00      1.00   

ItemID  VR_SV012  
count     389.00  
mean        0.63  
std         0.48  
min         0.00  
25%         0.00  
50%         1.00  
75%         1.00  
max         1.00  

[8 rows x 195 columns]

In [212]:
##proportion of people who got it right (sum of those values == 1 divided by N)
proportionRight =VR_Score.describe().round(2).iloc[1]

In [213]:
proportionRight =proportionRight.iloc[1:]

In [214]:
cp

ItemID        PIN                        RegistrationID  VA_RA022  VR_MR001  \
0             009  ABCE632D-1895-4DD5-9773-E848E3A1B396       NaN       NaN   
1        12345678  BE7A8565-9699-42D5-BA5F-86D304B50BA4       NaN       NaN   
2         5NA6010  8EE1F715-6AE0-43EC-B053-1EAB3BF9E1D0       NaN       NaN   
3        AL Train  E0DA4239-102F-4B79-9775-86DE3C1BD6BE       NaN       NaN   
4       ATLCERT11  1E28145A-D99E-4FBF-B253-D21EB86445EA       NaN       NaN   
...           ...                                   ...       ...       ...   
3495     msl50001  E51171E2-F00C-4B48-885B-5A08C6E5190C       NaN       NaN   
3496     msl50005  E2E3E3EE-D085-4513-85C5-8EFE10EA9AEE       NaN       NaN   
3497     sna40104  B22BD1C5-C330-44FA-96F7-41CF7425970D       NaN       NaN   
3498     wPX22103  AC5AC37B-4AB4-433A-BAB7-073B36DAD54F       NaN       NaN   
3499     wpx22106  F215A453-F8D7-48B6-8FD1-E767D12CFC41       NaN       NaN   

ItemID  VR_MR002  VR_MR003  VR_MR004  VR_MR005  VR_MR006  VR_MR007  ...  \
0            NaN       NaN       NaN       NaN       NaN       NaN  ...   
1            NaN       NaN       NaN  26.12822       NaN       NaN  ...   
2            NaN       NaN       NaN       NaN       NaN       NaN  ...   
3            NaN       NaN       NaN       NaN       NaN       NaN  ...   
4            NaN       NaN       NaN       NaN       NaN       NaN  ...   
...          ...       ...       ...       ...       ...       ...  ...   
3495         NaN       NaN       NaN       NaN       NaN       NaN  ...   
3496    41.97112       NaN       NaN  30.48706       NaN       NaN  ...   
3497         NaN       NaN       NaN       NaN       NaN       NaN  ...   
3498         NaN       NaN  36.76374       NaN       NaN       NaN  ...   
3499         NaN       NaN       NaN       NaN       NaN       NaN  ...   

ItemID  VR_SV003  VR_SV004R1  VR_SV005R1  VR_SV006R1  VR_SV007R1  VR_SV008R1  \
0            NaN         NaN         NaN         NaN         NaN         NaN   
1            NaN         NaN         NaN    13.73057         NaN         NaN   
2            NaN         NaN         NaN         NaN         NaN         NaN   
3            NaN         NaN         NaN         NaN         NaN         NaN   
4            NaN    10.90024         NaN         NaN         NaN         NaN   
...          ...         ...         ...         ...         ...         ...   
3495         NaN         NaN         NaN         NaN         NaN         NaN   
3496         NaN         NaN         NaN         NaN    30.39164         NaN   
3497         NaN         NaN    6.997987         NaN         NaN         NaN   
3498         NaN    17.12378         NaN         NaN         NaN         NaN   
3499         NaN    16.52227   14.548000         NaN         NaN         NaN   

ItemID  VR_SV009  VR_SV010  VR_SV011  VR_SV012  
0            NaN       NaN       NaN  3.605265  
1            NaN       NaN       NaN       NaN  
2            NaN       NaN  27.20981       NaN  
3            NaN       NaN       NaN       NaN  
4            NaN       NaN       NaN       NaN  
...          ...       ...       ...       ...  
3495         NaN       NaN       NaN       NaN  
3496         NaN       NaN       NaN       NaN  
3497         NaN       NaN       NaN       NaN  
3498         NaN       NaN       NaN       NaN  
3499         NaN       NaN       NaN       NaN  

[3500 rows x 196 columns]

In [215]:
#within those people who got it right, what was their mean 
meantime_c = cp.describe().round(2).iloc[1]
#within those people who got it right, what was their SD
sdtime_c = cp.describe().round(2).iloc[2]

In [216]:
correctcol = pd.concat([proportionRight,meantime_c, sdtime_c], axis=1)

In [217]:
correctcol.columns = ['ProportionCorrect', 'MeanTime_Correct', 'SDTime_Correct']

In [218]:
datacorrect =pd.merge(data,correctcol, how='left',on="ItemID")

In [219]:
cp[cp['VR_SV008R1'].notnull()]

ItemID       PIN                        RegistrationID  VA_RA022  VR_MR001  \
218     MAP60015  FF15E5E2-E345-4F09-9C59-AFE005F8D4BC       NaN       NaN   
413     MCB19001  C0E34076-15F1-495F-8D78-B68C61CC972C       NaN       NaN   
425     MCB20101  247F294C-7976-4AEA-80F5-BC8570B8036E       NaN       NaN   
464     MCB40003  FE317D93-81D1-4B0B-B9F7-256B6871080D       NaN       NaN   
489     MCB50106  81814267-AF32-4FA9-93F9-D54B9F892A32       NaN       NaN   
516     MCB70117  F250C87C-5310-4EE1-9246-378B5355E50C       NaN  26.53662   
601     MCH12017  C470A222-F2B2-4E83-A0FF-FC200098D97B       NaN       NaN   
643     MCH17012  E376F57B-9491-4A52-BD87-2242C40232B2       NaN       NaN   
708     MCH50001  D610C529-512F-4398-BFE8-47DB65EF68EE       NaN       NaN   
885     MSL20101  F63C19DD-B516-4B8F-AA4C-E38E8252D1D4       NaN  33.71593   
898     MSL22101  DBFBD593-A8B6-4CB1-9AAF-3074ED0C2934       NaN       NaN   
909     MSL30106  836DC54F-2F08-4829-818F-625174E0F70B       NaN       NaN   
916     MSL40103  09F17EF9-322C-4B10-B552-58D7EBF6EC1B       NaN       NaN   
950     MSL70102  429C2ECA-4453-46DA-8750-C05E82A6531D       NaN       NaN   
1127    NBS17011  337C5955-4736-47D4-9CD4-876169EB10C4       NaN       NaN   
1409    NNJ16020  03B2A876-15C9-41A0-ACE4-68468815D4A1       NaN       NaN   
1675    SAT12010  AF62236A-4E7A-422C-B21C-F4BBF7A052F7       NaN       NaN   
1728    SAT16011  4E278E75-350A-4032-9AED-F12770859FC0       NaN  11.35632   
1758    SAT20003  ECB23F6C-6519-43D0-80A4-F236283491CC       NaN       NaN   
1792    SAT30116  EC5F340C-2EE4-43FD-9A14-5E5EE05711A2       NaN       NaN   
1842    SAT70117  D84A3D0A-073A-4882-963A-A1BC759D087F       NaN       NaN   
1951    SBM13009  4A4A865E-2DFC-4A95-860A-77D65D6EFADB       NaN       NaN   
2047    SBM30111  7CB978A9-9CFA-45AD-BFD5-2D100970879F       NaN       NaN   
2063    SBM40111  67252702-9BF2-40FB-8C7E-987F1FAE7D70       NaN       NaN   
2068    SBM40119  AA631143-3764-47D0-8A80-63F5776DDD46       NaN       NaN   
2085    SBM50103  630A65A8-211A-47B6-958D-D837008E492F       NaN       NaN   
2221    SDA11008  BAA946BC-63A6-45AC-840B-978D3EFA0F1B       NaN       NaN   
2241    SDA12020  9ED68390-1FCA-4639-AF69-F11BC5DA22A4       NaN       NaN   
2293    SDA15020  EA698676-0C86-4E0B-BC4C-CB40B9F4FE93       NaN       NaN   
2326    SDA19003  35A482DE-0DAE-4F5C-A2DE-82144AC7CC67       NaN       NaN   
2603    SNA15016  1C808EC2-30DB-4C22-96CD-FD134579C4B6       NaN       NaN   
2667    SNA40005  FAA21B5C-26A5-44C2-A4A3-E3E18403948D       NaN       NaN   
2754    Sna60152  74E9666A-2631-4BBD-9848-77CACC606B89       NaN       NaN   
2958    WLA16023  8A09A926-F8E9-4B9B-9B9B-A3696E488AFF       NaN       NaN   
2994    WLA19003  C3FBE66C-8FB5-4233-86C3-A072422462C8       NaN       NaN   
3076    WLA40126  2354B6C3-5D84-44D3-B7E9-FE847238E539       NaN       NaN   
3205    WPX13015  ED4B2723-C778-43F8-860F-F8DCCE758EF9       NaN       NaN   
3213    WPX14014  DD118C9A-AD8A-464B-977D-868769AE0C7A       NaN       NaN   
3226    WPX16011  4FA75E8B-DD61-49AE-BD24-C2935766C49D       NaN       NaN   
3249    WPX20110  29F972F0-99FB-4D08-A5E3-98BA732AF726       NaN       NaN   
3288    WPX40120  9CE41A68-FF44-42D6-A74B-49E1E09448C0       NaN       NaN   
3390    Wpx13007  8C3EEE46-524F-479A-80EC-6DB39B376DB8       NaN       NaN   
3420    Wpx17002  5532BC40-0ED2-4465-9AFE-CAE49BED4224       NaN       NaN   
3468    Wpx60007  CBCC2002-05B0-4E8E-8C0E-B08959198A5A       NaN       NaN   

ItemID  VR_MR002  VR_MR003  VR_MR004   VR_MR005  VR_MR006   VR_MR007  ...  \
218          NaN       NaN       NaN        NaN       NaN        NaN  ...   
413          NaN       NaN       NaN        NaN       NaN        NaN  ...   
425          NaN       NaN  21.17172        NaN       NaN        NaN  ...   
464          NaN       NaN       NaN        NaN       NaN        NaN  ...   
489          NaN       NaN       NaN        NaN       NaN        NaN  ...   
516          NaN       NaN 

In [220]:
#within those people who got it wrong, what was their mean 
meantime_w = wp.describe().round(2).iloc[1]
#within those people who got it wrong, what was their SD
sdtime_w = wp.describe().round(2).iloc[2]

In [221]:
wcol = pd.concat([meantime_w, sdtime_w], axis=1)

In [222]:
wcol.columns = [ 'MeanTime_Incorrect', 'SDTime_Incorrect']

In [223]:
final =pd.merge(datacorrect, wcol, how='left',on="ItemID")

In [224]:
final.shape

(194, 9)

In [225]:
common = final.merge(vr1['ItemID'],on=['ItemID'], how="right")

In [226]:
common.drop_duplicates()

ItemID  mean_age      N  MeanTime_Overall  ProportionCorrect  \
0     VR_RA058R1     25.40  598.0              8.92               0.58   
1    VR_SS3004R1      7.68  188.0              5.98               0.18   
2       VR_PC002      7.71  192.0              5.11               0.77   
3    VR_SS3005R1     24.48  614.0              8.36               0.77   
4     VR_RA062R1     24.07  608.0             10.93               0.50   
..           ...       ...    ...               ...                ...   
684     VR_RA001      3.52  112.0              3.88               0.46   
688   VR_RA044R1      3.50  111.0              4.02               0.50   
710   VR_RA011R1      3.56  113.0              4.84               0.25   
805   VR_MR025R1     32.80  387.0             16.60               0.50   
815  VR_SS2111R1     33.30  393.0              8.00               0.46   

     MeanTime_Correct  SDTime_Correct  MeanTime_Incorrect  SDTime_Incorrect  
0                8.94            4.29                8.69              6.23  
1                6.15            3.38                6.29              3.92  
2                5.46            2.42                3.82              2.73  
3                8.34            4.90                8.41              6.72  
4               11.55            6.95               10.18              6.84  
..                ...             ...                 ...               ...  
684              4.54            3.82                2.97              4.55  
688              4.34            3.95                3.82              4.55  
710              3.83            1.77                5.10             10.21  
805             18.11           10.90               15.16             11.63  
815              8.56            4.21                7.58              6.74  

[194 rows x 9 columns]

In [227]:
final[final.ItemID == 'VR_PC013']

ItemID  mean_age      N  MeanTime_Overall  ProportionCorrect  \
50  VR_PC013      7.99  177.0              4.85               0.77   

    MeanTime_Correct  SDTime_Correct  MeanTime_Incorrect  SDTime_Incorrect  
50              5.12            2.03                3.96              2.35

In [228]:
final.to_csv('C:/Users/tsb7592/Downloads/Reaction_time.csv', index=False)

In [229]:
final.shape

(194, 9)